src/lib/InstantSearch.ts

```diff
@@ -587,6 +588,20 @@ Feel free to give us feedback on GitHub: https://github.com/algolia/instantsearc
 
     this.mainHelper.clearCache().search();
   }
+
+  public sendEventToInsights(event: InsightsEvent) {
```

Comment:
this probably should be called directly from the widgets, without a builtin method for insights in the InstantSearch API

Reply:
`instantSearchInstance.sendEventToInsights()` will be mostly called from connectors.
(For example, [here](https://github.com/algolia/instantsearch.js/blob/feat/send-events/src/connectors/refinement-list/connectRefinementList.js#L308-L312) -> [here](https://github.com/algolia/instantsearch.js/blob/feat/send-events/src/lib/utils/createSendEventForFacet.ts#L19:L19))

And we can guide users to use it when they want to send a custom event.
So I've exposed `sendEventToInsights()` in InstantSearch as a public method.
What do you think?

Reply:
It makes sense, this way the widgets can assume the middleware exists, even if it doesn't. 

We have a similar thing with createURL returning a default of `#`

I thought a little, but can't think of a way to share this default value efficiently. The downside I see here is that if someone wants to make their own middleware which is optional, they need to add some code to InstantSearch

Reply:
Could we maybe do something like `instantsearchInstance.sendEvent?()` (which gets called in the widget directly, after being added to the instance by the middleware)

Reply:
> The downside I see here is that if someone wants to make their own middleware which is optional, they need to add some code to InstantSearch

I agree. Potentially we could have `instantSearchInstant.middlewareBlahBlah` as a hub to access middlewares or things returned/added by middlewares. I think you left this idea as a comment somewhere in this PR or the other.

For now, I like your idea [here](https://github.com/algolia/instantsearch.js/pull/4456#discussion_r462103344). I'll try that.